### ISSUE TITLE
Make Chat in trade UI more visible

### ISSUE THREAD

**Issue by HenrikJannsen** [bisq-network/bisq2#1501]
The chat window is below the trade process view and will be only visible when scrolling on most screens.
We should find a better solution to make it easier to recognize. Maybe an extra button with icon at the trade process header would help? This button could open a new window (same as detach).

Maybe also rename the "Detach" button to "Undock" .

**Comment by axpoems** [comment]
I can address this, as a follow-up to the other issues related to the Bisq Easy UI and the new chat design.

**Comment by HenrikJannsen** [comment]
I added milestone 2.0.0 to give that higher priority, though I would not consider it a showstopper if we don't get it done for release...

**Comment by axpoems** [comment]
This proposal presents a new design concept for the 'My Open Trades' page. It introduces two viewing modes: one that presents a comprehensive list of open trades, and another that condenses the list into a left-side column format. In the condensed view, the right side features a set of windows providing detailed information on the selected trade from the column list. This set includes a header with trade specifics and a window displaying the protocol process alongside the trade chat. This layout improves visibility, particularly for the trade chat. Overall, this design provides quicker access and enhanced visibility to multiple open trades information, especially for users managing numerous trades.

<br>

Detailed list view:
![my-open-trades-1](https://github.com/bisq-network/bisq2/assets/145597137/b8c63f4c-33af-4c3f-8028-f5a036978835)

<br>


Condensed column list and specific trade view:
![my-open-trades-2](https://github.com/bisq-network/bisq2/assets/145597137/4f77c325-6546-4be7-a4af-e035cc992bd8)

<br>
@HenrikJannsen what do you think?

**Comment by HenrikJannsen** [comment]
Interesting idea. 
I fear the trade details input fields (or part of it) might be not visible in small screen size.
When the list is removed from top, the trade chat might be less hidden as before. Also if we keep the "detach" feature the user can move it to an new window. I think if we keep the layout as it was but find a way to make the chat "detach" button more visible it would be better.
Alternatively we could add a tab view for the trade state view and the chat. But not sure if that is a good idea.

The arrow for the left navigation version and the details version might not be that clear as its not a typical UX feature.
Maybe a icon for different  views at the header would be better?
I assume the collapsed view is the default.  For most users and our target user, the newby as buyer, i assume its usually only one trade at a time. In that case we could omit the option so switch views.

**Comment by djing-chan** [comment]
I removed milestone assignment as its not considered a showstopper for release.

**Comment by axpoems** [comment]
@HenrikJannsen  - Thanks for you inputs!

> I fear the trade details input fields (or part of it) might be not visible in small screen size.

Good observation. Since we don't necessarily need to have all the fields there as one single row, some could go on the main header for instance, like trade id (as already is) with a few others, for example me and peer. Then we can have two lines of info an make sure everything is visible.
Also since the list holds info as well, we could avoid repetition from there as well.

---

> When the list is removed from top, the trade chat might be less hidden as before. Also if we keep the "detach" feature the user can move it to an new window. I think if we keep the layout as it was but find a way to make the chat "detach" button more visible it would be better.

The 'My Open Trades' tab needs to be more dynamic, particularly because it comprises three information components, which may not display adequately on the majority of medium to small screens. I've created this proposal to tackle the issue you've highlighted. Currently, as you continue adding trades, you reach a stage where both the chat and the trade process window goes out of screen.
To address this, one possibility is that the blocks of information on this page (trade list, trade process, and trade chat) could be presented in different tabs as you mention. However, this approach would miss the point of providing all three components visible simultaneously. Again by organizing the components in a more dynamic layout, users can seamlessly access all aspects of their trades, even on smaller screens. This serves users with one or many trades, but those with multiple trades get most of the benefit.

---

> Alternatively we could add a tab view for the trade state view and the chat. But not sure if that is a good idea.

I don't think that's a good idea, for the reasons I outline above.

---

> The arrow for the left navigation version and the details version might not be that clear as its not a typical UX feature.
> Maybe a icon for different views at the header would be better?

Yes, the arrow was merely included to show the concept in the mock-up. Your suggestion is a good solution, and also with a proper transition, we could fit a better icon or design for it.
I have seen that Spotify has a similar layout concept as the one I am proposing in this mock-up (I saw this after proposing this). If you want to experiment, just use Spotify. For this feature in particular, it's the "Your library" column.

---

> I assume the collapsed view is the default. For most users and our target user, the newbie as buyer, i assume its usually only one trade at a time. In that case we could omit the option so switch views.

I had initially thought the list as the default view, with the ability to open specific trades and transition to the multiple window layout. However, the reverse approach could also work. Additionally, I'm considering adding a second collapse option for the trade list column, which could be convenient in a lack of screen space or the scenario for new users with only one trade as you say. (See also Spotify - they actually have this option as well. And again, I confirmed this after the proposal).

---

To summarize, my perspective is that this proposal enhances visibility and interoperability among the three main components: the open trade list, the trade process, and the chat. And I believe the way to think about it is this:

1. List -------- 2. Trade process --------- 3. Chat

All of them have the nature of a column.
With trade process being the most important, and the list extra-collapsed/collapsed/expanded as needed by user needs. Then chat follows.

**Comment by HenrikJannsen** [comment]
Maybe we can extract the trade step indicator into a horizontal bar below the trade details spanning both form and chat columns? 

But I am not sure if 2 columns next to each other gives the impression that both have equal priority. Users might miss that they have to use the trade step view to proceed in the process and using the chat alone will not trigger any progress.
Having the chat below would give a natural order of priority. With having the list remove from the top we have less of the space problems. We also could consider to pack the details bar into the header bar...

Just rough inputs...

**Comment by axpoems** [comment]
> Maybe we can extract the trade step indicator into a horizontal bar below the trade details spanning both form and chat columns?

Yes I was contemplating it as well. The content of the trade process in this proposal would still have more improvements than what is shown. I haven't focus on this at this stage.

> But I am not sure if 2 columns next to each other gives the impression that both have equal priority. Users might miss that they have to use the trade step view to proceed in the process and using the chat alone will not trigger any progress.

There's no way to give it significant visibility/usability without bring it up (beside the trade process). The detach feature is as good as having it in another tab, not a good UX IMO. 
Maybe we need to highlight more the action in the trade process so the user is directed to know what to do next. Also, other options could be a smaller size to the chat (40/60) or that it could be collapsed, in the latter it's existence would be visible but only usable/open when needed.
And if the user doesn't know what to do then maybe that's also what the chat is for, the peer is there to give him some guidance/instructions.

> Having the chat below would give a natural order of priority. With having the list remove from the top we have less of the space problems. We also could consider to pack the details bar into the header bar...

I understand that and that's more or less a space optimization of what we have now. For what I remember when I tested options, it wouldn't accomplish/have enough space to have a good usability of the chat, especially in smaller screens.
However, experimenting having these two components one on top of the other or one next to the other is a simple change we can test directly on the app and will also give us the best idea in terms of UX.

---

Summarising our discussion so far:
1. Explore setting the trade steps horizontally (perhaps included in the main header) and bringing the current process step higher so that it's the first thing to be seen.
1. Improve even more the 'trade details header' so that all the important info is visible (even in smaller screens).
1. Make the chat window narrower (i.e. smaller width than trade process). Consider making it collapsible.
1. Improve the trade list usability so it is easy to understand how to collapse/expand.

Let me know your thoughts.

**Comment by HenrikJannsen** [comment]
Sounds good! Allowing the chat to collapse might add the risk that the user miss messages from the peer... We could let the user change the width distribution of process and chat view by using a split pane. Though JavaFX SplitPane implementation is not great.

**Comment by HenrikJannsen** [comment]
FYI the current version has a max height for the table ( i guess 3 rows), so the initial state problem that the table eats up too much space is at least a bit mitigated.
